app/assets/javascripts/govuk_publishing_components/lib/single-consent-functions.js

```diff
@@ -0,0 +1,65 @@
+/* global GovSingleConsent */
+// = require govuk-single-consent/dist/singleconsent.iife.js
+
+(function (root) {
+  'use strict'
+  window.GOVUK = window.GOVUK || {}
+
+  window.GOVUK.singleConsent = {
+    init: function (callback) {
+      if (!this.useConsentApi()) {
+        return
+      }
+      callback = callback || this.apiCallBack
+      // determine where we are and set the consent api URL accordingly
+      if (!this.url) {
+        this.url = 'staging'
+        var environment = window.GOVUK.loadAnalytics.getEnvironment(window.GOVUK.analyticsGa4.core.trackFunctions.getHostname())
+        if (environment) {
+          this.url = environment.consentApiUrl
+        }
+      }
+      // create the consent API object
+      this.consentApiObj = new GovSingleConsent(callback, this.url)
+    },
+
+    useConsentApi: function () {
+      if (window.GOVUK.useSingleConsentApi) {
+        return true
```

Comment:
I think I've massively over-complicated this, and only actually need to refer to the original `window` variable. I was thinking it would be good to have a single place where this was handled (in the consent library) and then refer to that, but it's just gotten way too complex. Have updated.